notebooks/py/MERFISH.py

```diff
@@ -0,0 +1,252 @@
+# at least we can do it with get_slice and set_slice right now.
+from copy import deepcopy
+filtered_imgs = deepcopy(low_passed)
+
+for selector in imgs._iter_axes():
+    data = filtered_imgs.get_slice(selector)[0]
+    scaled = data / scale_factors[selector[Axes.ROUND.value], selector[Axes.CH.value]]
+    filtered_imgs.set_slice(selector, scaled, [Axes.ZPLANE])
+# EPY: END code
+
+# EPY: START markdown
+### Visualize processed data
+# EPY: END markdown
+
+# EPY: START code
+single_plane_filtered = filtered_imgs.sel({Axes.ROUND: 0, Axes.CH: 0, Axes.ZPLANE: 0})
+single_plane_filtered = single_plane_filtered.xarray.squeeze()
+
+plt.figure(figsize=(10,10))
+plt.subplot(121)
+plt.imshow(single_plane, cmap='gray', clim = list(np.percentile(single_plane.data, [5, 99])))
+plt.axis('off')
+plt.title('Original data, Round:0, Channel: 0')
+plt.subplot(122)
+plt.imshow(single_plane_filtered, cmap='gray', clim = list(np.percentile(single_plane_filtered.data, [5, 99])))
+plt.title('Filtered data, Round:0, Channel: 0')
+plt.axis('off');
+# EPY: END code
+
+# EPY: START markdown
+### Decode the processed data into spatially resolved gene expression profiles
+#
+#Here, we decode each pixel value, across all rounds and channels into the corresponding target (gene) it corresponds too. Contiguous pixels that map to the same target gene are called as one RNA molecule. Intuitively, pixel vectors are matched to the codebook by computing the euclidean distance between the pixel vector and all codewords. The minimal distance gene target is then selected. 
+# EPY: END markdown
+
+# EPY: START code
+from starfish.spots import PixelSpotDecoder
+
+psd = PixelSpotDecoder.PixelSpotDecoder(
+    codebook=experiment.codebook,
+    metric='euclidean', # distance metric to use for computing distance between a pixel vector and a codeword
+    norm_order=2, # the L_n norm is taken of each pixel vector and codeword before computing the distance. this is n
+    distance_threshold=0.5176, # minimum distance between a pixel vector and a codeword for it to be called as a gene
+    magnitude_threshold=1.77e-5, # discard any pixel vectors below this magnitude
+    min_area=2, # do not call a 'spot' if it's area is below this threshold (measured in pixels)
+    max_area=np.inf, # do not call a 'spot' if it's area is above this threshold (measured in pixels)
+)
+
+initial_spot_intensities, prop_results = psd.run(filtered_imgs)
+
+spot_intensities = initial_spot_intensities.loc[initial_spot_intensities[Features.PASSES_THRESHOLDS]]
+# EPY: END code
+
+# EPY: START markdown
+### Compare to results from paper
+#
+#The below plot aggregates gene copy number across single cells in the field of view and compares the results to the published intensities in the MERFISH paper. Note that Starfish detects a lower number of transcripts than the authors' results. This can probably be improved by tweaking any of the above parameters.
```

Comment:
```suggestion
#The below plot aggregates gene copy number across single cells in the field of view and compares the results to the published counts in the MERFISH paper. Note that Starfish detects a lower number of transcripts than the authors' results. This can likely be improved by tweaking the parameters of the algorithms above.
```